### PR TITLE
Fix map metadata how to

### PR DESCRIPTION
**Pull request by nabobalis**
I created a how to guide and copied the basic example. 

Maybe we want there to be more infomation or a more complex example?

### CODE REVIEW

File: docs/how_to/fix_map_metadata.rst [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+.. code-block:: python
+
+    >>> from astropy.io import fits
+
+    >>> from sunpy.map import Map
+    >>> import sunpy.data.sample
+
+    >>> filepath = sunpy.data.sample.AIA_171_IMAGE  # doctest: +REMOTE_DATA
+    >>> data, header = fits.getdata(filepath, header=True)  # doctest: +REMOTE_DATA
+    # Note that it is case insensitive for the keys
```
Comment (nabobalis) [inline docs/how_to/fix_map_metadata.rst:21]: is this true? I think it is not.

File: docs/how_to/fix_map_metadata.rst [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+.. code-block:: python
+
+    >>> from astropy.io import fits
+
+    >>> from sunpy.map import Map
+    >>> import sunpy.data.sample
+
+    >>> filepath = sunpy.data.sample.AIA_171_IMAGE  # doctest: +REMOTE_DATA
+    >>> data, header = fits.getdata(filepath, header=True)  # doctest: +REMOTE_DATA
+    >>> # Note that it is case insensitive for the keys
```
Comment (nabobalis) [inline docs/how_to/fix_map_metadata.rst:21]: I added this but I am pretty sure its wrong.

File: docs/reference/map.rst [hunk 3]
```diff
@@ -88,60 +41,10 @@ The header_helper sub-module contains helper functions for generating FITS-WCS h
...
+Defined in ``sunpy.map.sources`` are a set of `~sunpy.map.GenericMap` subclasses which convert the specific metadata and other differences in each instruments data to the standard `~sunpy.map.GenericMap` interface.
+These 'sources' also define things like the colormap and default normalization for each instrument.
+These subclasses also provide a method, which describes to the `~sunpy.map.Map` factory which data and metadata pairs match its instrument.
 
 .. automodapi:: sunpy.map.sources
     :no-main-docstr:
     :inherited-members:
-
-
-Writing a new Instrument Map Class
```
Comment (nabobalis) [inline docs/reference/map.rst:104]: This is already in a topic guide and that topic guide is more advanced.

File: docs/how_to/fix_map_metadata.rst [hunk 4]
```diff
@@ -0,0 +1,28 @@
+.. _sunpy-how-to-fix-map-metadata:
+
+*************************
+Fixing incorrect metadata
+*************************
+
+There will be times where you will come across a FITS files with either incorrect, missing or unparsable metadata and reading these files into `~sunpy.map.Map` will cause an error.
+Therefore, to load these files into a `~sunpy.map.Map`, you will need to correct the metadata before hand.
```
Comment (wtbarnes) [inline docs/how_to/fix_map_metadata.rst:8]:
```suggestion
Therefore, to load these files into a `~sunpy.map.Map`, you will need to correct the metadata beforehand.
```

File: docs/how_to/fix_map_metadata.rst [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+
+This will heavily depend on what metadata requires changing, unforutanlly it is near impossible to have a guide for every possible keyword.
+Below we have an example where the units in the FITS header are incorrect, this is controlled by the ``cunit1`` and ``cunit2`` keywords in a FITS header.
+So we will change those values and then load the file into a `~sunpy.map.Map`.
```
Comment (wtbarnes) [inline docs/how_to/fix_map_metadata.rst:12]:
```suggestion
In the example below, the units in the FITS header,  as controlled by the ``CUNIT1`` and ``CUNIT2`` keywords, are incorrect.
Before loading the file into a `~sunpy.map.Map`, we will correct these keywords to have the correct units.
```

File: docs/how_to/fix_map_metadata.rst [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+    >>> import sunpy.data.sample
+
+    >>> filepath = sunpy.data.sample.AIA_171_IMAGE  # doctest: +REMOTE_DATA
+    >>> data, header = fits.getdata(filepath, header=True)  # doctest: +REMOTE_DATA
+    >>> # Note that it is case insensitive for the keys
+    >>> header['cunit1'] = 'arcsec'  # doctest: +REMOTE_DATA
+    >>> header['cunit2'] = 'arcsec'  # doctest: +REMOTE_DATA
+    >>> updated_map = Map(data, header)  # doctest: +REMOTE_DATA
+
+This applies for any FITS standard keyword, which you can find in the `FITS standard <https://fits.gsfc.nasa.gov/fits_standard.html>`__.
```
Comment (wtbarnes) [inline docs/how_to/fix_map_metadata.rst:28]:
```suggestion
This applies for any keyword in the `FITS standard <https://fits.gsfc.nasa.gov/fits_standard.html>`__.
```

File: docs/reference/map.rst [hunk 7]
```diff
@@ -5,61 +5,14 @@ Maps (`sunpy.map`)
...
-class `~sunpy.map.Map`: ::
-
-    >>> x = sunpy.map.Map('file.fits')  # doctest: +SKIP
-
-The result of a call to `~sunpy.map.Map` will be either a `~sunpy.map.mapbase.GenericMap` object,
-or a subclass of `~sunpy.map.mapbase.GenericMap` which either deals with a specific type of data,
-e.g. `~sunpy.map.sources.sdo.AIAMap` or `~sunpy.map.sources.soho.LASCOMap`
-(see :ref:`map-classes` to see a list of all of them), or if no
-instrument matches, a 2D map `~sunpy.map.mapbase.GenericMap`.
-
```
Comment (wtbarnes) [inline docs/reference/map.rst:46]: Is this information captured somewhere else?

**Review by wtbarnes — CHANGES_REQUESTED** [review]

File: docs/reference/map.rst (hunk 7 above)
Comment (nabobalis) [reply docs/reference/map.rst:46]: Err, I do not think it is but for me I didn't think it adds much but I can restore.

Comment (nabobalis) [reply docs/reference/map.rst:46]: I added it back.

File: docs/how_to/fix_map_metadata.rst [hunk 8]
```diff
@@ -0,0 +1,27 @@
+.. _sunpy-how-to-fix-map-metadata:
+
+*************************
+Fixing incorrect metadata
+*************************
+
+There will be times where you will come across a FITS files with either incorrect, missing or unparsable metadata and reading these files into `~sunpy.map.Map` will cause an error.
+Therefore, to load these files into a `~sunpy.map.Map`, you will need to correct the metadata beforehand.
+
+In the example below, the units in the FITS header,  as controlled by the ``CUNIT1`` and ``CUNIT2`` keywords, are incorrect.
```
Comment (wtbarnes) [inline docs/how_to/fix_map_metadata.rst:10]:
```suggestion
In the example below, the units in the FITS header, as controlled by the ``CUNIT1`` and ``CUNIT2`` keywords, are incorrect.
```

File: docs/reference/map.rst [hunk 9]
```diff
@@ -5,66 +5,22 @@ Maps (`sunpy.map`)
...
 
+.. currentmodule:: sunpy.map
+
 sunpy.map Package
 =================
 
+Map objects are constructed using the special factory class: `~sunpy.map.Map`.
 All sunpy Maps are derived from `sunpy.map.GenericMap`, all the methods and attributes are documented in that class.
 
+The result of a call to `~sunpy.map.Map` will be either a `~sunpy.map.mapbase.GenericMap` object if no instrument matches, or a subclass of `~sunpy.map.mapbase.GenericMap` which deals with a specific source of data, e.g., `~sunpy.map.sources.sdo.AIAMap` or `~sunpy.map.sources.soho.LASCOMap`(see :ref:`map-classes` to see a list of all of them).
```
Comment (wtbarnes) [inline docs/reference/map.rst:22]: Is this line not too long?

**Review by wtbarnes — APPROVED** [review]

File: docs/reference/map.rst (hunk 9 above)
Comment (nabobalis) [reply docs/reference/map.rst:22]: Very long. But it's one sentence :P

**Review by alasdairwilson — APPROVED** [review]
Lgtm

**Pull request merged**